src/confluent_kafka/src/Admin.c

```diff
@@ -4219,6 +4309,57 @@ Admin_c_GroupResults_to_py (const rd_kafka_group_result_t **c_result_responses,
         return all_groups_result;
 }
 
+/**
+ *
+ * @brief Convert C ListOffsetsResultInfo array to dict[TopicPartition, ListOffsetsResultInfo].
+ *
+ */
+static PyObject *Admin_c_ListOffsetsResultInfos_to_py (const rd_kafka_ListOffsetsResultInfo_t **c_result_infos, size_t c_result_info_cnt) {
+        PyObject *result = NULL;
+        PyObject *ListOffsetsResultInfo_type = NULL;
+        size_t i;
+
+        ListOffsetsResultInfo_type = cfl_PyObject_lookup("confluent_kafka.admin",
+                                                         "ListOffsetsResultInfo");
+        if(!ListOffsetsResultInfo_type){
+                return NULL;
+        }
+
+        result = PyDict_New();
+        for(i=0; i<c_result_info_cnt; i++){
+                PyObject *value = NULL;
+                const rd_kafka_topic_partition_t *c_topic_partition = rd_kafka_ListOffsetsResultInfo_topic_partition(c_result_infos[i]);
+
+                int64_t c_timestamp = rd_kafka_ListOffsetsResultInfo_timestamp(c_result_infos[i]);
+
+                if (c_topic_partition->err) {
+                        value = KafkaError_new_or_None(c_topic_partition->err,rd_kafka_err2str(c_topic_partition->err));
+                } else {
+                        PyObject *args = NULL;
+                        PyObject *kwargs = NULL;
+                        kwargs = PyDict_New();
+                        cfl_PyDict_SetLong(kwargs,"offset", c_topic_partition->offset);
+                        cfl_PyDict_SetLong(kwargs,"timestamp", c_timestamp);
+                        cfl_PyDict_SetInt(kwargs,"leader_epoch",
+                                rd_kafka_topic_partition_get_leader_epoch(c_topic_partition));
+                        args = PyTuple_New(0);
+                        value = PyObject_Call(ListOffsetsResultInfo_type, args, kwargs);
+                        Py_DECREF(args);
+                        Py_DECREF(kwargs);
+                        if (value == NULL)
+                                goto raise;
+                }
+                PyDict_SetItem(result, c_part_to_py(c_topic_partition), value);
+                Py_DECREF(value);
+        }
+
+        Py_DECREF(ListOffsetsResultInfo_type);
+        return result;
+raise:
+        Py_DECREF(result);
```

Comment:
Use `Py_XDECREF` instead for raise cases.

when `i=0`, `result=NULL`, this statement will give segfault. Its safer to use XDECREF in general.

Reply:
result cannot be NULL here, it's initialized before the for loop and there's no goto before

Reply:
Still I prefer XDECREF. But its fine this way as well.